bot/cogs/queue.py

```diff
@@ -178,78 +161,65 @@ async def join(self, ctx):
+            queue = [self.bot.get_user(user_id) for user_id in queue_ids]
+            guild_data = await self.bot.db_helper.get_guild(ctx.guild)
+            capacity = guild_data['capacity']
 
-            if ctx.author in queue.active:  # Author already in queue
+            if ctx.author in queue:  # Author already in queue
                 title = f'**{ctx.author.display_name}** is already in the queue'
-            elif len(queue.active) >= queue.capacity:  # Queue full
+            elif len(queue) >= capacity:  # Queue full
                 title = f'Unable to add **{ctx.author.display_name}**: Queue is full'
             elif not player:  # ApiHelper couldn't get player
                 title = f'Unable to add **{ctx.author.display_name}**: Cannot verify match status'
             elif player.in_match:  # User is already in a match
                 title = f'Unable to add **{ctx.author.display_name}**: They are already in a match'
             else:  # User can be added
-                queue.active.append(ctx.author)
+                await self.bot.db_helper.insert_queued_users(ctx.guild, ctx.author)
+                queue += [ctx.author]
                 title = f'**{ctx.author.display_name}** has been added to the queue'
 
                 # Check and burst queue if full
-                if len(queue.active) == queue.capacity:
+                if len(queue) == capacity:
                     try:
-                        all_readied = await self.start_match(ctx, queue.active)
+                        all_readied = await self.start_match(ctx, queue)
                     except asyncio.TimeoutError:
                         return
 
                     if all_readied:
-                        queue.bursted = queue.active  # Save bursted queue for player draft
-                        queue.active = []  # Reset the player queue to empty
+                        await self.bot.db_helper.delete_queued_users(ctx.guild, *queue)
 
                     return
 
-        # Send message based on outcome
-        embed = self.queue_embed(ctx.guild, title)
+        embed = await self.queue_embed(ctx.guild, title)
 
         # Delete last queue message
-        if queue.last_msg:
-            try:
-                await queue.last_msg.delete()
-            except discord.errors.NotFound:
-                pass
-
-        queue.last_msg = await ctx.send(embed=embed)
+        await self.update_last_msg(ctx, embed)
 
     @commands.command(brief='Leave the queue (or the bursted queue)')
     async def leave(self, ctx):
         """ Check if the member can be remobed from the guild and remove them if so. """
-        queue = self.guild_queues[ctx.guild]
+        removed = await self.bot.db_helper.delete_queued_users(ctx.guild, ctx.author)
 
-        if ctx.author in queue.active:
-            queue.active.remove(ctx.author)
+        if ctx.author.id in removed:
             title = f'**{ctx.author.display_name}** has been removed from the queue '
```

Comment:
Is the space at the end of the string a requirement?

Reply:
Don't think so.